spark/src/main/scala/org/apache/iceberg/spark/SparkTableUtil.scala

```diff
@@ -297,5 +301,81 @@ object SparkTableUtil {
       )
     }
   }
+
+  private def buildManifest(table: Table,
+                    sparkDataFiles: Seq[SparkDataFile],
+                    partitionSpec: PartitionSpec): ManifestFile = {
+    val outputFile = table.io
+      .newOutputFile(FileFormat.AVRO.addExtension("/tmp/" + UUID.randomUUID.toString))
+    val writer = ManifestWriter.write(partitionSpec, outputFile)
+    try {
+      for (file <- sparkDataFiles) {
+        writer.add(file.toDataFile(partitionSpec))
+      }
+    } finally {
+      writer.close()
+    }
+
+    writer.toManifestFile
+  }
+
+  /**
+   * Import a spark table to a iceberg table.
+   *
+   * The import uses the spark session to get table metadata. It assumes no
+   * operation is going on original table and target table and thus is not
+   * thread-safe.
+   *
+   * @param source the database name of the table to be import
+   * @param location the location used to store table metadata
+   *
+   * @return table the imported table
+   */
+  def importSparkTable(source: TableIdentifier, location: String): Table = {
+    val sparkSession = SparkSession.builder().getOrCreate()
+    import sparkSession.sqlContext.implicits._
+
+    val dbName = source.database.getOrElse("default")
+    val tableName = source.table
+
+    if (!sparkSession.catalog.tableExists(dbName, tableName)) {
+      throw new NoSuchTableException(s"Table $dbName.$tableName does not exist")
+    }
+
+    val partitionSpec = SparkSchemaUtil.specForTable(sparkSession, s"$dbName.$tableName")
+    val conf = sparkSession.sparkContext.hadoopConfiguration
+    val tables = new HadoopTables(conf)
+    val schema = SparkSchemaUtil.schemaForTable(sparkSession, s"$dbName.$tableName")
+    val table = tables.create(schema, partitionSpec, ImmutableMap.of(), location)
+    val appender = table.newAppend()
+
+    if (partitionSpec == PartitionSpec.unpartitioned) {
```

Comment:
Why is there a special case for unpartitioned tables? Doesn't `partitionDF` return a single partition in that case?

Reply:
Yes, here is the exception:
```
org.apache.hadoop.hive.ql.metadata.HiveException: Table item is not a partitioned table
  at org.apache.hadoop.hive.ql.metadata.Hive.getPartitions(Hive.java:2123)
  at org.apache.hadoop.hive.ql.metadata.Hive.getPartitions(Hive.java:2156)
  at org.apache.spark.sql.hive.client.HiveClientImpl$$anonfun$getPartitions$1.apply(HiveClientImpl.scala:670)
  at org.apache.spark.sql.hive.client.HiveClientImpl$$anonfun$getPartitions$1.apply(HiveClientImpl.scala:662)
  at org.apache.spark.sql.hive.client.HiveClientImpl$$anonfun$withHiveState$1.apply(HiveClientImpl.scala:275)
  at org.apache.spark.sql.hive.client.HiveClientImpl.liftedTree1$1(HiveClientImpl.scala:213)
  at org.apache.spark.sql.hive.client.HiveClientImpl.retryLocked(HiveClientImpl.scala:212)
  at org.apache.spark.sql.hive.client.HiveClientImpl.withHiveState(HiveClientImpl.scala:258)
  at org.apache.spark.sql.hive.client.HiveClientImpl.getPartitions(HiveClientImpl.scala:662)
  at org.apache.spark.sql.hive.client.HiveClient$class.getPartitions(HiveClient.scala:210)
  at org.apache.spark.sql.hive.client.HiveClientImpl.getPartitions(HiveClientImpl.scala:83)
  at org.apache.iceberg.spark.hacks.Hive.partitions(Hive.java:48)
  at org.apache.iceberg.spark.SparkTableUtil$.partitionDF(SparkTableUtil.scala:56)
```

If that is not expected, I can try to fix that first